src/p2p/NodeList.ts

```diff
@@ -305,13 +305,33 @@ export function removeNodes(
   for (const id of ids) removeNode(id, raiseEvents, cycle)
 }
 
+// create shorthand type for node with refuteCycles
+export type NodeWithRefuteCycles = P2P.NodeListTypes.Node & { refuteCycles: Set<number> }
+
 export function updateNode(
   update: P2P.NodeListTypes.Update,
   raiseEvents: boolean,
   cycle: P2P.CycleCreatorTypes.CycleRecord | null
 ) {
-  const node = nodes.get(update.id)
+  const node = nodes.get(update.id) as NodeWithRefuteCycles
   if (node) {
+    // Initialize refuteCycles if it doesn't exist
+    if (!node.refuteCycles) {
```

Comment:
I think this should be in the if statement below as it can impact the hash of a node

Reply:
or at least inside of config.p2p.enableProblematicNodeRemoval